src/glmtools.jl

```diff
@@ -20,57 +180,157 @@ end
 function mueta{T<:Real}(::CloglogLink, η::T)
     max(eps(T), exp(η) * exp(-exp(η)))
 end
+function inverselink(::CloglogLink, η)
+    expη = exp(η)
+    μ = -expm1(-expη)
+    omμ = exp(-expη)   # the complement, 1 - μ
+    μ, max(realmin(μ), expη * omμ), max(realmin(μ), μ * omμ)
+end
 
 linkfun(::IdentityLink, μ) = μ
 linkinv(::IdentityLink, η) = η
-mueta(::IdentityLink, η) = 1
+mueta(::IdentityLink, η) = one(η)
+inverselink(::IdentityLink, η) = η, one(η), oftype(η, NaN)
 
 linkfun(::InverseLink, μ) = inv(μ)
 linkinv(::InverseLink, η) = inv(η)
 mueta(::InverseLink, η) = -inv(abs2(η))
+function inverselink(::InverseLink, η)
+    μ = inv(η)
+    μ, -abs2(μ), oftype(μ, NaN)
+end
+
+linkfun(::InverseSquareLink, μ) = inv(abs2(μ))
+linkinv(::InverseSquareLink, η) = inv(sqrt(η))
```

Comment:
Ok, I've looked more carefully at the inverse Gaussian case now. The thing is that there is a bit of freedom in how you can choose in the canonical parameters in the inverse Gaussian model. You can e.g. set `θ = 1/μ²` or `θ = -1/(2μ²)` which have different signs. The former choice implies a negative variance function but therefore also a negative dispersion. The latter version has positive variance function and dispersion but `θ<0`.

However, it seems that you can scale the variance function with any (non-zero) constant and it will still work since it will end up as the same factor in the score and the information so it will cancel in the update. When the variance is calculated, the estimation of the dispersion will capture the "wrong" factor. Bottomline is that a fast version should also work just fine even if we keep R's choices of functions. However, it is still a bit confusing that the functions don't follow the theory here.